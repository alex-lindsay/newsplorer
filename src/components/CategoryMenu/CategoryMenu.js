import React from "react";
import styles from "./CategoryMenu.module.css";

import IconButton from "@material-ui/core/IconButton";
import Badge from "@material-ui/core/Badge";
import CategoryIcon from "@material-ui/icons/Category";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";

const CategoryMenu = props => {
  const [anchorEl, setAnchorEl] = React.useState(null);
  const open = Boolean(anchorEl);

  const handleMenu = event => {
    setAnchorEl(event.currentTarget);
  };
  const handleClose = event => {
    setAnchorEl(null);
    props.setCategory(event.currentTarget.dataset.category);
  };

  const menuItems = props.categories
    ? props.categories.sort().map(category => (
        <MenuItem
          key={category}
          onClick={handleClose}
          data-category={category}
          data-categoryid={category}
        >
          {category}
        </MenuItem>
      ))
    : null;
  return (
    <div className={styles.CategoryMenu} data-testid="category-menu">
      <Badge
        badgeContent={
          props.category ? props.category.substring(0, 3) + "â€¦" : null
        }
        anchorOrigin={{ vertical: "bottom", horizontal: "right" }}
        color="secondary"
        overlap="circle"
      >
        <IconButton
          aria-label="account of current user"
          aria-controls="menu-categorymenu"
          aria-haspopup="true"
          onClick={handleMenu}
          color="inherit"
        >
          <CategoryIcon />
        </IconButton>
      </Badge>
      <Menu
        id="menu-categorymenu"
        anchorEl={anchorEl}
        anchorOrigin={{
          vertical: "top",
          horizontal: "right",
        }}
        keepMounted
        transformOrigin={{
          vertical: "top",
          horizontal: "right",
        }}
        open={open}
        onClose={handleClose}
      >
        <MenuItem
          key="-"
          onClick={handleClose}
          data-category=""
          data-categoryid=""
        >
          None
        </MenuItem>
        {menuItems}
      </Menu>
    </div>
  );
};

export default CategoryMenu;
