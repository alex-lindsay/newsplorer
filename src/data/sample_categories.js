import { sampleSources } from "./sample_sources";

export const sampleCategories = Array.from(
  new Set(sampleSources["sources"].map(source => source.category))
);
