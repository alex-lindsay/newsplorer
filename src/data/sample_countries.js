import { sampleSources } from "./sample_sources";

export const sampleCountries = Array.from(
  new Set(sampleSources["sources"].map(source => source.country))
);
