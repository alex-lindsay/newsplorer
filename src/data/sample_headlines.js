export const sampleHeadlines = {
  status: "ok",
  totalResults: 38,
  articles: [
    {
      source: {
        id: "cnn",
        name: "CNN",
      },
      author: "Sara Ashley O'Brien, CNN Business",
      title: "Instacart makes more changes ahead of planned worker strike  CNN",
      description:
        "As some Instacart workers prepare to strike Monday, the company is making more updates in a bid to appease them.",
      url:
        "https://www.cnn.com/2020/03/29/tech/instacartstrikehandsanitizertips/index.html",
      urlToImage:
        "https://cdn.cnn.com/cnnnext/dam/assets/200323132443instacartdoordeliverysupertease.jpg",
      publishedAt: "20200329T21:44:16Z",
      content: null,
    },
    {
      source: {
        id: null,
        name: "Nypost.com",
      },
      author: "Reuters",
      title:
        "Coronavirus deaths fall again in Italy but lockdown extension looms  New York Post ",
      description:
        "ROME — The number of deaths from coronavirus in Italy fell for the second consecutive day on Sunday but the country still looked almost certain to see an extension of stringent containment me…",
      url:
        "https://nypost.com/2020/03/29/coronavirusdeathsfallagaininitalybutlockdownextensionlooms/",
      urlToImage:
        "https://thenypost.files.wordpress.com/2020/03/italyhospital.jpg?quality=90&strip=all&w=1200",
      publishedAt: "20200329T21:31:32Z",
      content:
        "ROME — The number of deaths from coronavirus in Italy fell for the second consecutive day on Sunday but the country still looked almost certain to see an extension of stringent containment measures.\r\nThe Civil Protection department said 756 people had died in… [+1032 chars]",
    },
    {
      source: {
        id: "cnn",
        name: "CNN",
      },
      author: "Chandelis Duster, CNN",
      title:
        "Cuomo: No one is going to treat New York 'unfairly' over coronavirus  CNN",
      description:
        'Gov. Andrew Cuomo had a message for the nation on Sunday: Although New York has become the epicenter of the Covid19 outbreak, no one will treat the state "unfairly" over coronavirus.',
      url:
        "https://www.cnn.com/2020/03/29/politics/andrewcuomocoronavirusbriefing/index.html",
      urlToImage:
        "https://cdn.cnn.com/cnnnext/dam/assets/200327135746governorcuomo3272020supertease.jpg",
      publishedAt: "20200329T21:12:58Z",
      content:
        'Washington (CNN)Gov. Andrew Cuomo had a message for the nation on Sunday: Although New York has become the epicenter of the Covid19 outbreak, no one will treat the state "unfairly" over coronavirus.\r\n"New York is going to have what it needs. And no one is go… [+3351 chars]',
    },
    {
      source: {
        id: null,
        name: "Gamespot.com",
      },
      author: "Justin Clark",
      title:
        "The Original Nier Is Returning To Consoles As Nier Replicant Ver. 1.22474487139  GameSpot",
      description:
        "The original Nier Is set to rerelease on PC, PS4, Xbox One with some big upgrades.",
      url:
        "https://www.gamespot.com/articles/theoriginalnierisreturningtoconsolesasnier/11006475348/",
      urlToImage:
        "https://gamespot1.cbsistatic.com/uploads/screen_kubrick/1544/15449755/3647866nierrep.jpg",
      publishedAt: "20200329T20:58:00Z",
      content:
        "To celebrate the 10th anniversary for the original Nier, SquareEnix held a 10hour livestream which included a concert performance and a chat with the game's creative leads. During that conversation, the surprise announcement was made, via a beautiful, but v… [+2046 chars]",
    },
    {
      source: {
        id: "foxnews",
        name: "Fox News",
      },
      author: "Associated Press",
      title: "Timmy Hill wins virtual Texas NASCAR race  Fox News",
      description: "The stars came out in Texas.",
      url: "https://www.foxnews.com/auto/timmyhillvirtualtexasnascarrace",
      urlToImage:
        "https://static.foxnews.com/foxnews.com/content/uploads/2020/03/timmy.jpg",
      publishedAt: "20200329T20:44:10Z",
      content:
        "Bob Weir of the Grateful Dead sang the national anthem from a remote location. Troy Aikman, borrowing Matthew McConaughey's trademark phrase “Alright, alright, alright,” commanded drivers to start their, uh, simulators.\r\nNASCAR's new normal resumed Sunday wit… [+3966 chars]",
    },
    {
      source: {
        id: null,
        name: "Variety.com",
      },
      author: "Chris Willman",
      title:
        "Joe Diffie, ’90s Country Music Star, Dies of Coronavirus at 61  Variety",
      description:
        "Joe Diffie, an icon to many country fans for his string of No. 1 hits in the 1990s, has died from complications related to the coronavirus, a spokesperson revealed Sunday afternoon. He was 61. “Grammywinning country music legend Joe Diffie passed away today,…",
      url:
        "https://variety.com/2020/music/news/joediffiecoronaviruscountrystardieddead1203548295/",
      urlToImage:
        "https://pmcvariety.files.wordpress.com/2020/03/shutterstock_editorial_10595814be1585342384562.jpg?w=700&h=393&crop=1",
      publishedAt: "20200329T20:23:20Z",
      content:
        "Joe Diffie, an icon to many country fans for his string of No. 1 hits in the 1990s, has died from complications related to the coronavirus, a spokesperson revealed Sunday afternoon. He was 61.\r\n“Grammywinning country music legend Joe Diffie passed away today… [+5486 chars]",
    },
    {
      source: {
        id: "businessinsider",
        name: "Business Insider",
      },
      author: "Paulina Cachero",
      title:
        "California Governor says US government sent 170 'broken' ventilators  Business Insider  Business Insider",
      description:
        'Instead of "complaining," California Governor Gavin Newsom said "broken" ventilators were sent to a local company to be repaired.',
      url:
        "https://www.businessinsider.com/californiagovernorsaysusgovernmentsent170brokenventilators20203",
      urlToImage:
        "https://i.insider.com/5e80ea242ff8395e140fed74?width=1200&format=jpeg",
      publishedAt: "20200329T20:20:15Z",
      content:
        "As hospitals across the country lack the critical medical supplies they need to battle the coronavirus, the Trump administration has been sending ventilators to states whose health care systems are straining in the midst of the outbreak.\r\nCalifornia, which ha… [+3511 chars]",
    },
    {
      source: {
        id: null,
        name: "Latimes.com",
      },
      author: "Joseph Serna",
      title:
        "'Grim Sleeper' serial killer Lonnie Franklin dies on death row  Los Angeles Times",
      description:
        "Lonnie David Franklin Jr., the Grim Sleeper serial killer, has died while on death row.",
      url:
        "https://www.latimes.com/california/story/20200329/grimsleeperserialkillerlonniefranklindiesdeathrowsanquentin",
      urlToImage:
        "https://catimes.brightspotcdn.com/dims4/default/b45d109/2147483647/strip/true/crop/2048x1075+0+60/resize/1200x630!/quality/90/?url=https%3A%2F%2Fcaliforniatimesbrightspot.s3.amazonaws.com%2F0f%2Fcb%2Fd92c220d82ac39e912e022d42337%2Fla1470861964snapphoto",
      publishedAt: "20200329T20:19:36Z",
      content:
        "Lonnie David Franklin Jr., the socalled Grim Sleeper serial killer who preyed on Los Angeles women for more than two decades, died Saturday while on death row at San Quentin prison, officials said.\r\nFranklin was found unresponsive in his single cell on March… [+2225 chars]",
    },
    {
      source: {
        id: "foxnews",
        name: "Fox News",
      },
      author: "Joseph Wulfsohn",
      title:
        "Trump tells Prince Harry, Meghan Markle 'they must pay' for security amid reported move to US  Fox News",
      description:
        "President Trump on Sunday didn't offer the warmest greeting to Prince Harry and Meghan Markle amid reports that they had relocated from Canada to the U.S.",
      url:
        "https://www.foxnews.com/politics/trumpprinceharrymeghanmarklemustpaysecuritymoveus",
      urlToImage:
        "https://static.foxnews.com/foxnews.com/content/uploads/2020/03/PrinceHarryMeghanTrump_AP.jpg",
      publishedAt: "20200329T20:02:40Z",
      content:
        "Get all the latest news on coronavirus and more delivered daily to your inbox.  Sign up here.\r\nPresident Trump on Sunday didn't offer the warmest greeting to Prince Harry and Meghan Markle amid reports that they had relocated from Canada to the U.S.\r\nA source… [+1487 chars]",
    },
    {
      source: {
        id: "usatoday",
        name: "USA Today",
      },
      author: "Hannah Yasharoff",
      title:
        "CBS mourns longtime journalist Maria Mercader, who died at 54 from coronavirus  USA TODAY",
      description:
        "CBS News is mourning Maria Mercader, a longtime journalist with the network who died in a New York hospital after battling coronavirus. She was 54.",
      url:
        "https://www.usatoday.com/story/entertainment/tv/2020/03/29/coronaviruscbsnewsmariamercaderdiescovid19age54/2936246001/",
      urlToImage:
        "https://www.gannettcdn.com/presto/2020/03/29/USAT/73c680977b194cdd800bb7019c90ac0aimage002.png?crop=954,537,x0,y128&width=1600&height=800&fit=bounds",
      publishedAt: "20200329T19:50:39Z",
      content:
        "Robert De Niro, Ben Stiller and La La Anthony join Blake Shelton, Kevin Bacon, Neil Diamond and more in telling people to practice social distancing.\r\nUSA TODAY\r\nCBS News is mourning Maria Mercader, a longtime journalist with the news outlet who died in a New… [+2999 chars]",
    },
    {
      source: {
        id: null,
        name: "Baltimoresun.com",
      },
      author: "Jessica Anderson",
      title:
        "Man in his 90s dies of coronavirus after outbreak at Mount Airy nursing home, Carroll Co. officials say  Baltimore Sun",
      description:
        "A man in his 90s passed away at the Pleasant View Nursing Home in Carroll County and 11 other residents have been taken to two local hospitals, Carroll County officials announced Sunday.",
      url:
        "https://www.baltimoresun.com/coronavirus/bsmdnursinghomedeath20200329aufiwfjmbzb57oxr37ylxrti2ustory.html",
      urlToImage:
        "https://www.baltimoresun.com/resizer/SL2D_CPpYuFJWhLlBNkXRAV5mN0=/1200x0/top/arcanglerfisharc2prodtronc.s3.amazonaws.com/public/566WM3P3CJCMPOS4EQFFGVRINE.jpg",
      publishedAt: "20200329T19:40:58Z",
      content: null,
    },
    {
      source: {
        id: null,
        name: "Clickondetroit.com",
      },
      author: "Cassidy Johncox",
      title:
        "Southeast Michigan mall owner Taubman requires tenants to pay rent amid closures  WDIV ClickOnDetroit",
      description:
        "Southeast Michigan mall owner Taubman Centers is requiring its tenants to pay rent during the coronavirus (COVID19) pandemic, according to a memo obtained by CNBC.",
      url:
        "https://www.clickondetroit.com/news/local/2020/03/29/southeastmichiganmallownertaubmanrequirestenantstopayrentamidclosures/",
      urlToImage:
        "https://www.clickondetroit.com/resizer/ZKvl56QaaN0o6KhyAgdfKyobW7s=/1600x713/smart/filters:format(jpeg):strip_exif(true):strip_icc(true):no_upscale(true):quality(65)/arcanglerfisharc2prodgmg.s3.amazonaws.com/public/7KQRDVRS7ZE5FO5QDOVWVQ36NI.jpg",
      publishedAt: "20200329T19:33:59Z",
      content:
        "Southeast Michigan mall owner Taubman Centers is requiring its tenants to pay rent during the coronavirus (COVID19) pandemic, according to a memo obtained by CNBC.\r\nThe Bloomfield Hills real estate investment trust owns 26 shopping centers in the U.S. and As… [+2016 chars]",
    },
    {
      source: {
        id: null,
        name: "Nypost.com",
      },
      author: "Mark Moore",
      title:
        "Trump boasts White House briefing ratings in attack of 'Lamestream Media'  New York Post ",
      description:
        "President Trump touted the skyhigh ratings of his White House coronavirus briefings — claiming they are driving the media “crazy” — as he reminded his millions of Twitter f…",
      url:
        "https://nypost.com/2020/03/29/trumpboastswhitehousebriefingratingsinattackoflamestreammedia/",
      urlToImage:
        "https://thenypost.files.wordpress.com/2020/03/trumpreturnse1585509433834.jpg?quality=90&strip=all&w=1200",
      publishedAt: "20200329T19:32:43Z",
      content:
        "President Trump touted the skyhigh ratings of his White House coronavirus briefings — claiming they are driving the media “crazy” — as he reminded his millions of Twitter followers that another one is coming up Sunday afternoon.\r\n“Because the ‘Ratings’ of my… [+1249 chars]",
    },
    {
      source: {
        id: null,
        name: "Npr.org",
      },
      author: "",
      title:
        "Coronavirus Closures Helped Jonesboro Avoid Deaths, More Injuries From Tornado  Getaka.co.in",
      description:
        "The tornado injured at least 22 people and caused extensive damage to many properties. Officials attribute the relatively low number of casualties in part to social distancing and business shutdowns.",
      url:
        "https://www.npr.org/2020/03/29/823532336/tornadostrikesarkansascityofficialssaypandemicclosureskeptpeoplesafe",
      urlToImage:
        "https://media.npr.org/assets/img/2020/03/29/rts37i36_wide25894d20dde4a141b4f6e5541ff23a4a063ac802.jpg?s=1400",
      publishedAt: "20200329T19:29:42Z",
      content:
        "A screen grab from a social media video shows a tornado tearing through Jonesboro, Ark., on Saturday.\r\nSaid Said/Triples S Phone and Computer Repair/via REUTERS\r\nA tornado tore through nearly 20 miles of the city of Jonesboro, Ark., on Saturday evening, sever… [+2656 chars]",
    },
    {
      source: {
        id: "theverge",
        name: "The Verge",
      },
      author: "Kim Lyons",
      title:
        "Latest OnePlus 8 leaks reveal “Interstellar Glow” and other colors  The Verge",
      description:
        "A pair of leaks shows renders for the upcoming OnePlus 8 series that include a slate of new colors, including Interstellar Glow, Glacial Green and Onyx Black.",
      url:
        "https://www.theverge.com/2020/3/29/21198701/latestoneplus8leaksrevealinterstellarglowandothercolors",
      urlToImage:
        "https://cdn.voxcdn.com/thumbor/YgypLqT1KAj4lyO_Rf6ZfJf3T0Y=/0x271:1257x929/fitin/1200x630/cdn.voxcdn.com/uploads/chorus_asset/file/19857125/oneplus_interstellar_glow.png",
      publishedAt: "20200329T19:15:16Z",
      content:
        "The new phones are expected next month\r\nWinFuture.de/Roland Quandt\r\nA pair of leaks shows renders for the upcoming OnePlus 8 series that include a slate of new colors, including a purplishred one called Interstellar Glow. WinFuture and Ishan Agarwal both pos… [+872 chars]",
    },
    {
      source: {
        id: null,
        name: "Cbssports.com",
      },
      author: "",
      title:
        "Lions agree to terms with former Packers wide receiver Geronimo Allison, per report  CBS Sports",
      description: "The Lions add another target for Stafford",
      url:
        "https://www.cbssports.com/nfl/news/lionsagreetotermswithformerpackerswidereceivergeronimoallisonperreport/",
      urlToImage:
        "https://sportshub.cbsistatic.com/i/r/2019/11/11/f94416e8080c4920822606d800868d8a/thumbnail/1200x675/e64926da2c8ddc49a56092305ff601d0/geronimoallisonpackers.jpg",
      publishedAt: "20200329T19:08:54Z",
      content:
        "The Detroit Lions have two of the most productive wideouts in the NFL in Marvin Jones and Kenny Golladay, and they are reportedly adding another target for Matthew Stafford this offseason. According to ESPN's Adam Schefter, the Lions have agreed to terms with… [+1367 chars]",
    },
    {
      source: {
        id: "usatoday",
        name: "USA Today",
      },
      author: "John Fritze",
      title:
        "Fauci predicts up to 200,000 U.S. deaths as Trump weighs adjusting coronavirus guidelines. Here is what you should know.  USA TODAY",
      description:
        "As Trump considers easing social distancing guidelines, health expert Anthony Fauci says millions will contract coronavirus, up to 200,000 may die.",
      url:
        "https://www.usatoday.com/story/news/politics/2020/03/29/faucicoronaviruscouldhitmillionstrumpconsidersguidelines/2935804001/",
      urlToImage:
        "https://www.gannettcdn.com/presto/2020/03/26/USAT/82abd153aadd4ca89c6bed6882f2fe5aAP20085826579663.jpg?crop=4251,2392,x0,y425&width=3200&height=1680&fit=bounds",
      publishedAt: "20200329T18:53:42Z",
      content:
        "President Trump told reporters that certain governors are not appreciative enough of his administration's response to the coronavirus pandemic.\r\nUSA TODAY\r\nWASHINGTON  As President Donald Trump weighs whether to ease social distancing guidelines intended to s… [+6315 chars]",
    },
    {
      source: {
        id: "thewallstreetjournal",
        name: "The Wall Street Journal",
      },
      author: "Max Colchester, Alistair MacDonald",
      title:
        "Britain’s Creaking National Health System Gears Up for Coronavirus Crisis  The Wall Street Journal",
      description:
        "After years of belttightening, staterun health service will be tested, but experts say it has some advantages over its U.S. counterpart",
      url:
        "https://www.wsj.com/articles/britainscreakingnationalhealthsystemgearsupforcoronaviruscrisis11585479601",
      urlToImage: "https://images.wsj.net/im169494/social",
      publishedAt: "20200329T18:35:00Z",
      content:
        "LONDONA vast convention center in east London is being turned into a sprawling hospital that can handle up to 4,000 patients. Thousands of retired nurses and doctors are being drafted back to work. The British army is delivering protective clothing to dozens … [+201 chars]",
    },
    {
      source: {
        id: null,
        name: "Triblive.com",
      },
      author: "Patrick Varine",
      title:
        "Artist transforms Uptown tiger into 'Tiger King' portrait of Joe Exotic  TribLIVE",
      description:
        "Pop culture is a big part of Jeremy Raymer’s art, and he knows the value of having a finger on the pulse of the cultural zeitgeist. So when a friend mentioned the old tiger’s head, painted on a wall near the intersection of Moultrie Street and Tustin Street b…",
      url:
        "https://triblive.com/local/pittsburghallegheny/artisttransformsuptowntigerintotigerkingportraitofjoeexotic/",
      urlToImage:
        "https://triblive.com/wpcontent/uploads/2020/03/2508315_web1_ptrTigerKing2033020.jpg",
      publishedAt: "20200329T18:29:14Z",
      content:
        "TribLIVE's Daily and Weekly email newsletters deliver the news you want and information you need, right to your inbox.\r\nPop culture is a big part of Jeremy Raymer’s art, and he knows the value of having a finger on the pulse of the cultural zeitgeist.\r\nSo whe… [+2372 chars]",
    },
    {
      source: {
        id: null,
        name: "Venturebeat.com",
      },
      author: "James Falkoff, Converge",
      title:
        "4 things you need to understand about edge computing  VentureBeat",
      description:
        "Some aspects of edge computing have been overhyped, while others have gone under the radar. Here's a look at what's most important.",
      url:
        "https://venturebeat.com/2020/03/29/4thingsyouneedtounderstandaboutedgecomputing/",
      urlToImage:
        "https://venturebeat.com/wpcontent/uploads/2020/03/edge.jpg?w=1200&strip=all",
      publishedAt: "20200329T17:29:10Z",
      content:
        "Edge computing has claimed a spot in the technology zeitgeist as one of the topics that signals novelty and cuttingedge thinking. For a few years now, it has been assumed that this way of doing computing is, one way or another, the future. But until recently… [+13364 chars]",
    },
  ],
};
