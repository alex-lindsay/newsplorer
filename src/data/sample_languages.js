import { sampleSources } from "./sample_sources";

export const sampleLanguages = Array.from(
  new Set(sampleSources["sources"].map(source => source.language))
);
