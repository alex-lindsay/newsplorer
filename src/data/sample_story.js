export const sampleStory = {
  source: {
    id: null,
    name: "Huffpost.com",
  },
  author: "Cole Delbyck",
  title:
    "Tina Fey, Adam Sandler Return To 'SNL' To Honor Hal Willner With Musical Tribute - HuffPost",
  description:
    "The legendary music producer died this week with symptoms consistent with COVID-19.",
  url:
    "https://www.huffpost.com/entry/tina-fey-adam-sandler-return-to-snl-to-honor-hal-willner-with-musical-tribute_n_5e932164c5b6ca9a9f01f723",
  urlToImage:
    "https://img.huffingtonpost.com/asset/5e9336383000002903153f9c.png?ops=1200_630",
  publishedAt: "2020-04-12T16:25:37Z",
  content:
    "“Saturday Night Live” stars of past and present came together to close out the show’s first-ever “at home” edition with a stirring tribute to legendary sketch music producer Hal Willner. \r\nThe “SNL” veteran, who had scored the late-night comedy series since t… [+2303 chars]",
  topics: ["Saturday Night Live", "Hal Willner", "The SNL"],
};
