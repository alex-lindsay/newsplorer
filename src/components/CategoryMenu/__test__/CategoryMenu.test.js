import React from "react";
import { shallow, mount } from "enzyme";

import CategoryMenu from "../CategoryMenu";

import { sampleCategories } from "../../../data/sample_categories";
import { ExpansionPanelActions, MenuItem } from "@material-ui/core";

const mockSetCategory = jest.fn((event) => {
  // console.log(event);
  return event;
});

describe("CategoryMenu", () => {
  it("renders with no props", () => {
    const wrapper = mount(<CategoryMenu />);
  });

  it("renders specifed categories", () => {
    const wrapper = mount(<CategoryMenu categories={sampleCategories} />);
    expect(wrapper.exists("#menu-categorymenu"));
    expect(wrapper.find("#menu-categorymenu li")).toHaveLength(
      sampleCategories.length + 1
    );
    expect(wrapper.find("#menu-categorymenu li").first().text()).toBe("None");
    expect(wrapper.exists(".MuiBadge-badge"));
    expect(wrapper.find(".MuiBadge-badge").text()).toBe("");
  });

  it("renders the selected category", () => {
    const wrapper = mount(
      <CategoryMenu
        categories={sampleCategories}
        category={sampleCategories[3]}
      />
    );
    expect(wrapper.exists("#menu-categorymenu"));
    expect(wrapper.exists(".MuiBadge-badge"));
    expect(wrapper.find(".MuiBadge-badge").text()).toBe(
      sampleCategories[3].substring(0, 3) + "â€¦"
    );
  });

  it("sets the selected category when list item is clicked", () => {
    const wrapper = mount(
      <CategoryMenu
        categories={sampleCategories}
        category={null}
        setCategory={mockSetCategory}
      />
    );
    wrapper
      .find("#menu-categorymenu")
      .first()
      .find("li")
      .at(4)
      .simulate("click");
    expect(mockSetCategory).toHaveBeenCalled();
    // expects 3 because li[0] is the manually added "None"
    expect(mockSetCategory.mock.results[0].value).toBe(sampleCategories[3]);
  });

  it("shows/hides menu when icon is clicked", () => {
    const wrapper = mount(
      <CategoryMenu
        categories={sampleCategories}
        category={null}
        setCategory={mockSetCategory}
      />
    );
    const menu = wrapper.find("div#menu-categorymenu").first();
    const menuAttributes = menu.getDOMNode().attributes;

    expect(menuAttributes.getNamedItem("aria-hidden").value).toBe("true");

    wrapper.find("button").first().simulate("click");
    expect(menuAttributes.getNamedItem("aria-hidden")).toBeNull();

    wrapper.find(MenuItem).first().simulate("click");
    expect(menuAttributes.getNamedItem("aria-hidden").value).toBe("true");
  });
});
