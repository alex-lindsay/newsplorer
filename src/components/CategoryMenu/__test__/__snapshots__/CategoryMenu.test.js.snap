// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches expected snapshot 1`] = `
<DocumentFragment>
  <div
    class="CategoryMenu"
    data-testid="category-menu"
  >
    <span
      class="MuiBadge-root"
    >
      <button
        aria-controls="menu-categorymenu"
        aria-haspopup="true"
        aria-label="account of current user"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2l-5.5 9h11z"
            />
            <circle
              cx="17.5"
              cy="17.5"
              r="4.5"
            />
            <path
              d="M3 13.5h8v8H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <span
        class="MuiBadge-badge MuiBadge-anchorOriginBottomRightCircle MuiBadge-colorSecondary MuiBadge-invisible"
      />
    </span>
  </div>
</DocumentFragment>
`;
